### PR TITLE
funding source status bug fix

### PR DESCRIPTION
**Pull request by nehajain008**
Story [AB#13906](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/13906)
Task [AB#14018](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/14018)

### CODE REVIEW

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderItemDeleteSteps.cs [hunk 1]
```diff
@@ -10,14 +13,17 @@ internal sealed class OrderItemDeleteSteps
     {
         private readonly Request request;
         private readonly Settings settings;
+        private readonly OrderContext orderContext;
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderItemDeleteSteps.cs:16]: Due to the final approach we don't need the order context so we can delete this line.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderItemDeleteSteps.cs [hunk 2]
```diff
@@ -10,14 +13,17 @@ internal sealed class OrderItemDeleteSteps
...
 
         public OrderItemDeleteSteps(
             Request request,
-            Settings settings)
+            Settings settings,
+            OrderContext orderContext)
         {
             this.request = request;
             this.settings = settings ?? throw new ArgumentNullException(nameof(settings));
+            this.orderContext = orderContext ?? throw new ArgumentNullException(nameof(orderContext));
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderItemDeleteSteps.cs:26]: We'll need to delete this line as well.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderItemDeleteSteps.cs [hunk 3]
```diff
@@ -10,14 +13,17 @@ internal sealed class OrderItemDeleteSteps
...
         private readonly Request request;
         private readonly Settings settings;
+        private readonly OrderContext orderContext;
         private DeleteOrderItemRequest deleteOrderItemRequest;
 
         public OrderItemDeleteSteps(
             Request request,
-            Settings settings)
+            Settings settings,
+            OrderContext orderContext)
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderItemDeleteSteps.cs:22]: And this line.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderItemDeleteSteps.cs [hunk 4]
```diff
@@ -1,6 +1,9 @@
 ﻿using System;
 using System.Threading.Tasks;
+using FluentAssertions;
+using NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.Support;
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderItemDeleteSteps.cs:4]: Once we've removed the order context references we will be able to remove this line.

**Review by bjss-chris — CHANGES_REQUESTED** [review]
Looks good – just some redundant code in `OrderItemDeleteSteps` to remove.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderItemDeleteSteps.cs (hunk 1 above)
Comment (nehajain008) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderItemDeleteSteps.cs]: Deleted this line.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderItemDeleteSteps.cs (hunk 2 above)
Comment (nehajain008) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderItemDeleteSteps.cs]: Deleted this line.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderItemDeleteSteps.cs (hunk 3 above)
Comment (nehajain008) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderItemDeleteSteps.cs]: Deleted this line.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderItemDeleteSteps.cs (hunk 4 above)
Comment (nehajain008) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderItemDeleteSteps.cs]: Deleted this line.

**Review by bjss-chris — APPROVED** [review]

**Pull request merged**